cub/cmake/cub-config.cmake

```diff
@@ -27,16 +30,53 @@ function(_cub_declare_interface_alias alias_name ugly_name)
 endfunction()
 
 #
-# Setup targets
+# Setup some internal cache variables
 #
 
-_cub_declare_interface_alias(CUB::CUB _CUB_CUB)
 # Pull in the include dir detected by cub-config-version.cmake
 set(_CUB_INCLUDE_DIR "${_CUB_VERSION_INCLUDE_DIR}"
   CACHE INTERNAL "Location of CUB headers."
+  FORCE
 )
 unset(_CUB_VERSION_INCLUDE_DIR CACHE) # Clear tmp variable from cache
+
+if (${CMAKE_FIND_PACKAGE_NAME}_FIND_QUIETLY)
+  set(_CUB_QUIET ON CACHE INTERNAL "Quiet mode enabled for CUB find_package calls." FORCE)
+  set(_CUB_QUIET_FLAG "QUIET" CACHE INTERNAL "" FORCE)
+else()
+  set(_CUB_QUIET OFF CACHE INTERNAL "Quiet mode enabled for CUB find_package calls." FORCE)
+  set(_CUB_QUIET_FLAG "" CACHE INTERNAL "" FORCE)
+endif()
+
+#
+# Setup dependencies
+#
+
+if (NOT TARGET CUB::libcudacxx)
+  if (TARGET Thrust::libcudacxx)
+    # Prefer the same libcudacxx as Thrust, if available:
+    _cub_declare_interface_alias(CUB::libcudacxx _CUB_libcudacxx)
+    target_link_libraries(_CUB_libcudacxx INTERFACE Thrust::libcudacxx)
+  else()
+    if (NOT TARGET libcudacxx::libcudacxx)
+      find_package(libcudacxx ${cub_libcudacxx_version} CONFIG ${_CUB_QUIET_FLAG} REQUIRED
```

Comment:
If you prefer finding the version of libcudacxx over a local one, you should structure this as 2 searches.
This first search has the HINTS and NO_DEFAULT_PATH, with the second just being a regular `find_package`. That will ensure you use the packaged libcudacxx

Reply:
This just came up in a discussion with @senior-zero -- what do you think about adding a `find_package(CUDAToolkit)` call here and automatically checking for libcu++ in the CTK path? Any potential problems you can see with that?

Reply:
`find_package(CUDAToolkit)` requires CMake 3.17+ so that will require a bump of the minimum from 3.15 ( NVHPC users would need 3.20 since that brought in NVHPC support to CMake).

Note a deal breaker by any means but something you might need to consider.

Reply:
Fixed -- holding off on the CUDAToolkit check for now. We can add this later if needed.